webapp/src/ts/effects/contacts.effects.ts

```diff
@@ -0,0 +1,141 @@
+
+@Injectable()
+export class ContactsEffects {
+  private contactsActions;
+  private globalActions;
+
+  constructor(
+    private actions$: Actions,
+    private store: Store,
+    private contactViewModelGeneratorService: ContactViewModelGeneratorService,
+    private contactSummaryService: ContactSummaryService,
+    private tasksForContactService: TasksForContactService,
+    private targetAggregateService: TargetAggregatesService,
+    private translateService: TranslateService,
+  ) {
+    this.contactsActions = new ContactsActions(store);
+    this.globalActions = new GlobalActions(store);
+  }
+
+  selectContact = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(ContactActionList.selectContact),
+      filter(({ payload: { id } }) => !!id),
+      exhaustMap(({ payload: { id, silent } }) => {
+        if (!silent) {
+          this.globalActions.setLoadingShowContent(id);
+        }
+        return from(this.contactViewModelGeneratorService.getContact(id, { getChildPlaces: true, merge: false })).pipe(
+          map(model => this.contactsActions.setSelected(model)),
+          catchError(error => {
+            console.error('Error selecting contact', error);
+            return of(this.globalActions.unsetSelected());
+          }),
+        );
+      }),
+    );
+  }, { dispatch: false });
+
+  setSelected = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(ContactActionList.setSelected),
+      withLatestFrom(
+        this.store.pipe(select(Selectors.getSelectedContact))
+      ),
+      exhaustMap(([{ payload: { selected } }, selectedContact]) => {
+        const refreshing = (selectedContact && selectedContact.doc._id) === selected.id;
+        this.globalActions.settingSelected(refreshing);
+        this.contactsActions.setLoadingSelectedContact();
+        this.contactsActions.setContactsLoadingSummary(true);
+        this.globalActions.clearCancelCallback();
+        const options = { getChildPlaces: true };
+        const title = (selected.type && selected.type.name_key) || 'contact.profile';
+        this.globalActions.setTitle(this.translateService.instant(title));
+        return from(this.contactViewModelGeneratorService.loadChildren(selectedContact, options)).pipe(
+          map(children => {
+            return this.contactsActions.receiveSelectedContactChildren(children);
+          }),
+          catchError(error => {
+            console.error('Error fetching children', error);
+            return of(this.globalActions.unsetSelected());
+          }),
+        );
+      })
+    );
+  },{ dispatch: false });
+
+  receiveSelectedContactChildren = createEffect(() => {
+    return this.actions$.pipe(
+      ofType(ContactActionList.receiveSelectedContactChildren),
+      withLatestFrom(
+        this.store.pipe(select(Selectors.getSelectedContact)),
+        this.store.pipe(select(Selectors.getForms)),
+      ),
+      exhaustMap(([, selectedContact, forms]) => {
```

Comment:
Minor syntax nitpick... the empty first element here (and the other 2 cases later) strikes me as somewhat fragile. Is eslint enforcing that? Otherwise, how about this?

```suggestion
      exhaustMap(([_payload, selectedContact, forms]) => {
```
...with the underscore to indicate it is unused but still add clarity to others scanning the line and understanding the signature of the callback.

Reply:
Yes, it's an eslint thing because I am defining a variable that I do not use. Your solution results in the same error. I am looking into how I can define the effects differently so that I always make use of the payload.